### ISSUE TITLE
Sanitycheck hardware map integration failed with some tests timeout.

### ISSUE THREAD

**Issue by chen-png** [zephyrproject-rtos/zephyr#22777]
**Describe the bug**
use the sanitycheck hardware map option to run the whole tests directory in parallel on four boards:
reel_board, frdm_k64f, sam_e70_xplained, mimxrt1050_evk,
**there are some successful tests failed due to timeout(those tests should be passed if only run one board), so it can't ensure the pass rate.**

in the 26 failed tests:
1. some tests failed due to timeout
INFO    -  38/724 reel_board                tests/application_development/libcxx/application_development.cpp.libcxx.exceptions
FAILED Timeout  (device 60.464s) 
INFO    -  53/724 reel_board                tests/posix/common/portability.posix.common       
 FAILED Timeout (device 60.484s)
INFO    - 467/724 frdm_k64f                 tests/ztest/base/testing.ztest.verbose_2           
FAILED Timeout (device 60.430s)

for a example, 
if i run sanitycheck -p reel_board -T tests command on reel_board, the tests/application_development/libcxx/application_development.cpp.libcxx.exceptions can be passed,
however, if i run sanitycheck --hardware-map map4.yml -T tests/, the tests/application_development/libcxx/application_development.cpp.libcxx.exceptions will be failed on reel board due to timeout.

**To Reproduce**
1. connect four boards:  reel_board, frdm_k64f, sam_e70_xplained, mimxrt1050_evk
2. sanitycheck --generate-hardware-map map4.yml
3. sanitycheck --hardware-map map4.yml --device-testing -T tests/

**Screenshots or console output**
 INFO    - 724 test configurations selected, 996 configurations discarded due to filters.
INFO    - Adding tasks to the queue...

INFO    - 724/724 frdm_k64f      tests/lib/fdtable/libraries.os.fdtable       PASSED(device 2.674s)
**INFO    -  663 of 724   tests passed (96.23%),   26   failed, 35 skipped with   0 warnings in 2460.19 seconds**
**INFO    - In total 441 test cases were executed on 4 out of total 218 platforms (1.83%)**

Hardware distribution summary:

| Board            | ID                                               |   Counter |
|------------------|--------------------------------------------------|-----------|
| reel_board       | OSHW000032254e4500328003ab98000684d1000097969900 |       132 |
| sam_e70_xplained | ATML2637010000000000                             |       181 |
| frdm_k64f        | 0240020116B95E69EB47A3D1                         |       190 |
| mimxrt1050_evk   | 000000123456                                     |       182 |

**Environment (please complete the following information):**
 - OS: fedora28
 - Toolchain: zephyr-sdk-0.10.3
 - Commit ID: d24e5e520c

**Comment by nashif** [comment]
looks like a duplicate of #22780
